packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx

```diff
@@ -303,3 +268,181 @@ const localPreviewTrampoline = (mobileView: boolean, doNotShow?: boolean): JSX.E
+          onTroubleshootingClick={
+            onPermissionsTroubleshootingClick
+              ? () => {
+                  onPermissionsTroubleshootingClick(permissionsState);
+                }
+              : undefined
+          }
+          type="request"
+        />
+      );
+    }
+    // if audio permission is set up but video is not
+    else if (videoState === 'prompt') {
+      return (
+        <CameraDomainPermissions
+          appName={'app'}
+          onTroubleshootingClick={
+            onPermissionsTroubleshootingClick
+              ? () => {
+                  onPermissionsTroubleshootingClick(permissionsState);
+                }
+              : undefined
+          }
+          onContinueAnywayClick={() => {
+            setIsModalShowing(false);
+          }}
+          type="request"
+        />
+      );
+    }
+    // if video permission is set up but audio is not
+    else if (audioState === 'prompt') {
+      return (
+        <MicrophoneDomainPermissions
+          appName={'app'}
+          onTroubleshootingClick={
+            onPermissionsTroubleshootingClick
+              ? () => {
+                  onPermissionsTroubleshootingClick(permissionsState);
+                }
+              : undefined
+          }
+          type="request"
+        />
+      );
+    }
+    // if both video and audio are denied
+    else if (videoState === 'denied' && audioState === 'denied') {
+      return (
+        <CameraAndMicrophoneDomainPermissions
+          appName={'app'}
+          onTroubleshootingClick={
+            onPermissionsTroubleshootingClick
+              ? () => {
+                  onPermissionsTroubleshootingClick(permissionsState);
+                }
+              : undefined
+          }
+          type="denied"
+        />
+      );
+    }
+    // if only video is denied
+    else if (videoState === 'denied') {
+      return (
+        <CameraDomainPermissions
+          appName={'app'}
+          onTroubleshootingClick={
+            onPermissionsTroubleshootingClick
+              ? () => {
+                  onPermissionsTroubleshootingClick(permissionsState);
+                }
+              : undefined
+          }
+          onContinueAnywayClick={() => {
+            setIsModalShowing(false);
+          }}
+          type="denied"
+        />
+      );
+    }
+    // if only audio is denied
+    else if (audioState === 'denied') {
+      return (
+        <MicrophoneDomainPermissions
+          appName={'app'}
+          onTroubleshootingClick={
+            onPermissionsTroubleshootingClick
+              ? () => {
+                  onPermissionsTroubleshootingClick(permissionsState);
+                }
+              : undefined
+          }
+          type="denied"
+        />
+      );
+    }
+    // if neither state is denied or prompt, we don't need to return anything
+    else {
+      return undefined;
+    }
+  };
+
+  if (mobileView && modal() !== undefined) {
```

Comment:
I'm not sure what react does here when a react component is used in a conditional, my assumption is it has to perform the same render computation it would do in a render pass but my gut reaction is to avoid this

Something you can do that makes the code slightly more complicated is:
const 


But happy to have push back and leave this as is 
```tsx
const showModal = videoState === 'denied' || videoState === 'prompt || audioState === 'denied' || audioState === 'prompt;
const modal: undefined | () => JSX.Element = !showModal ? undefined : () => {
   ... modal code
}
```

So instead of a function that returns a react node or undefined, the function is undefined or it returns a react element

Reply:
made changes and pushed to PR https://github.com/Azure/communication-ui-library/pull/2529